### ISSUE TITLE
custom set: methods should act on el.innerHTML, not el.value, when in 'wysiwyg' mode

### ISSUE THREAD

**Issue by jywarren** [bevacqua/banksy#1]
This is for use with Woofmark's 'wysiwyg' mode, but could use an addition to the readme. I'm happy to write it if you're interested?

Essentially, if this isn't done, the chunk's before value is sometimes deleted when autocompleting, resulting in losing all text before the insertion point. (I'm disambiguating several different bugs, but this is my best understanding right now).

``` js
set: function (value) {
  if (woofmark.mode === 'wysiwyg') {
    woofmark.textarea.innerHTML = value;
  } else {
    woofmark.textarea.value = value;
  }
}
```

Thanks, great library!

**Comment by bevacqua** [comment]
Feels like woofmark needs better methods for setting its value, having the consumer deal with this is sloppy. Do you think you could come up with a fix at the `woofmark` API level?

**Comment by jywarren** [comment]
Yeah, i was thinking -- in the library I'm working on, wrapping woofmark, I
tried making getter/setter `markdown()` and `html()`, and considered making
`editor.value(mode, *value)` or something. Or variant with
`editor.value(*value)` defaulting to markdown when there's only one
parameter.

I definitely hear that the simplicity of markdown-only `value()` is nice.
Thoughts on keeping a simple API?

**Comment by bevacqua** [comment]
Can't we use the current `woofmark.value` method?

**Comment by jywarren** [comment]
Oh, that's what I meant -- but it currently only sets/gets in markdown. Although, is that a problem, here, as long as it parses cleanly into HTML?

**Comment by bevacqua** [comment]
Not sure I understand the problem here. Maybe write the failing tests first?

**Comment by jywarren** [comment]
OK, almost there on tests, but I'm still having trouble triggering the autocomplete via a crossvent trigger. If you have a moment, I'd be grateful for any tips you might have -- posted in more depth on #2 comments. Thanks!
